master/internal/api_master.go

```diff
@@ -25,3 +26,41 @@ func (a *apiServer) GetMaster(
 		Config:    configStruct,
 	}, err
 }
+
+func (a *apiServer) MasterLogs(
+	req *apiv1.MasterLogsRequest, resp apiv1.Determined_MasterLogsServer) error {
+	if err := grpc.ValidateRequest(
+		func() (bool, string) { return req.Limit >= 0, "Limit must be >= 0" },
```

Comment:
should we also validation `offset < a.m.logs.Len()` (sort of how we set offset = 0 if the negative offset was too large).

Reply:
I went back and forth on this. Because the offset may be valid in the future, all the logging endpoints will accept offset that are not yet valid.